packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -384,19 +384,31 @@ bool _hasMetadata(
       .isNotEmpty;
 }
 
-List<Error> _validateAst(Root root) {
+List<Error> _validateAst(Root root, String source) {
   final List<Error> result = <Error>[];
   final List<String> customClasses =
       root.classes.map((Class x) => x.name).toList();
   final List<String> customEnums = root.enums.map((Enum x) => x.name).toList();
   for (final Class klass in root.classes) {
     for (final Field field in klass.fields) {
-      if (!(_validTypes.contains(field.dataType) ||
+      if (field.dataType.contains('<')) {
+        result.add(Error(
+          message:
+              'Unsupported datatype:"${field.dataType}" in class "${klass.name}". Generic fields aren\'t yet supported (https://github.com/flutter/flutter/issues/63468).',
+          lineNumber: (field.offset != null)
+              ? _calculateLineNumber(source, field.offset!)
+              : null,
```

Comment:
Should _calculateLineNumber just accept null so you don't need to keep repeating this construction?

Reply:
I thought about that.  I ultimately decided against it because just considering `_calculateLineNumber`, I want non-null input and output.  It seemed to fly in the face of having nullable types to throw it out the window for the sake of ease of use.  I could also pull this into a helper function if I get sick of repeating it, that might be a better solution.  I can do that.